ArduPlane/sensors.cpp

```diff
@@ -98,6 +100,31 @@ void Plane::read_airspeed(void)
         if (airspeed.get_temperature(temperature)) {
             barometer.set_external_temperature(temperature);
         }
+
+        // check for airspeed hardware failure but wait until:
+        // a few seconds after a launch or until after an auto takeoff completes.
+        bool stages_to_ignore = (flight_stage == AP_SpdHgtControl::FLIGHT_TAKEOFF ||
+                flight_stage == AP_SpdHgtControl::FLIGHT_LAND_PREFLARE ||
+                flight_stage == AP_SpdHgtControl::FLIGHT_LAND_FINAL ||
+                flight_stage == AP_SpdHgtControl::FLIGHT_VTOL);
+
+        if (is_flying() &&
+            !(control_mode == AUTO && stages_to_ignore) &&
+            AP_HAL::millis() - started_flying_ms > 10000) {
+
+            bool airspeed_data_validity;
+
+            // perform airspeed hardware failure self-check. Fail when error > abs(target*25%)
+            float trashf; Vector2f trash2f; Vector3f trash3f;
+            float true_airspeed_variance;
+            if (ahrs.get_variances(trashf,trashf,trashf,trash3f, true_airspeed_variance, trash2f)) {
+                airspeed_data_validity = true_airspeed_variance < 1.0f;
+            } else {
+                const float error_threshold = target_airspeed_cm * 0.002f; // cmd to m conversion * 20% error.
```

Comment:
This check is far to strict for my aircraft at least. Unless I'm doing my math wrong, is there a way to graph the errors out of some logs? (Curious what it thinks the error is when the plane is fast to do a descent).